packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,416 @@
+        })
+        await txn.insertKey({ name: row.name, key })
+
+        await txn.remove({ category, name: row.name })
+        key.handle.free()
+        updateCount++
+      }
+      await txn.commit()
+    }
+
+    this.agent.config.logger.info(`Migrated ${updateCount} records of type ${category}`)
+  }
+
+  private async updateMasterSecret() {
+    if (!this.store) {
+      throw new IndySdkToAskarMigrationError(
+        'Update master secret can not be called outside of the `update()` function'
+      )
+    }
+
+    const category = 'Indy::MasterSecret'
+
+    this.agent.config.logger.info(`Migrating category: ${category}`)
+
+    let updateCount = 0
+    const session = this.store.transaction()
+
+    for (;;) {
+      const txn = await session.open()
+      const masterSecrets = await txn.fetchAll({ category, limit: 50 })
+      if (!masterSecrets || masterSecrets.length === 0) {
+        await txn.close()
+        break
+      }
+
+      if (!masterSecrets.some((ms: EntryObject) => ms.name === this.defaultLinkSecretId)) {
+        throw new IndySdkToAskarMigrationError('defaultLinkSecretId can not be established.')
+      }
+
+      this.agent.config.logger.info(`Default link secret id for migration is ${this.defaultLinkSecretId}`)
+
+      for (const row of masterSecrets) {
+        this.agent.config.logger.debug(`Migrating ${row.name} to the new askar format`)
+
+        const isDefault = masterSecrets.length === 0 ?? row.name === this.walletConfig.id
+
+        const {
+          value: { ms },
+        } = JSON.parse(row.value as string) as { value: { ms: string } }
+
+        const record = new AnonCredsLinkSecretRecord({ linkSecretId: row.name, value: ms })
+        record.setTag('isDefault', isDefault)
+        const value = JsonTransformer.serialize(record)
+
+        const tags = transformFromRecordTagValues(record.getTags())
+
+        await txn.insert({ category: record.type, name: record.id, value, tags })
+
+        await txn.remove({ category, name: row.name })
+        updateCount++
+      }
+      await txn.commit()
+    }
+
+    this.agent.config.logger.info(`Migrated ${updateCount} records of type ${category}`)
+  }
+
+  private async updateCredentials() {
+    if (!this.store) {
+      throw new IndySdkToAskarMigrationError('Update credentials can not be called outside of the `update()` function')
+    }
+
+    const category = 'Indy::Credential'
+
+    this.agent.config.logger.info(`Migrating category: ${category}`)
+
+    let updateCount = 0
+    const session = this.store.transaction()
+    for (;;) {
+      const txn = await session.open()
+      const credentials = await txn.fetchAll({ category, limit: 50 })
+      if (!credentials || credentials.length === 0) {
+        await txn.close()
+        break
+      }
+
+      for (const row of credentials) {
+        this.agent.config.logger.debug(`Migrating ${row.name} to the new askar format`)
+        const data = JSON.parse(row.value as string) as {
```

Comment:
how does this type differ from `AnonCredsCredential` interface? 

I see it has rev_reg and witness keys. Are those added when stored, or are those also transfered over the wire from issuer to holder?

Reply:
I see your interface contains the `rev_reg` and `witness` properties, but these are not in the `AnonCredsCredential`. Do you know if these properties are added when storing to the credential, or is the interface we have in AFJ incorrect (and we should add those keys to that interface)